cloudify_types/cloudify_types/component/component.py

```diff
@@ -230,6 +231,16 @@ def _set_secrets(self):
             return
 
         for secret_name in self.secrets:
+            existing_secret_value = get_secret_by_name(self.client,
```

Comment:
Can this perhaps be made a little more efficient by retrieving a list of all secret names before entering this loop and then just checking if secret_name in all_secrets?